ext/readline/readline.c

```diff
@@ -307,6 +315,22 @@ PHP_FUNCTION(readline_info)
 			RETVAL_STRING(SAFE_STRING(rl_prompt));
 		} else if (!strcasecmp(what, "terminal_name")) {
 			RETVAL_STRING((char *)SAFE_STRING(rl_terminal_name));
+		} else if (!strcasecmp(what, "completion_suppress_append")) {
+			oldval = rl_completion_suppress_append;
+			if (value) {
+				rl_completion_suppress_append = zend_is_true(value);
+			}
+			RETVAL_BOOL(oldval);
+		} else if (!strcasecmp(what, "completion_append_character")) {
+			char completion_append_str[2];
+			oldval = rl_completion_append_character;
+			if (value) {
+				convert_to_string_ex(value)
+				rl_completion_append_character = (int)Z_STRVAL_P(value)[0];
+			}
+			completion_append_str[0] = (char)oldval;
+			completion_append_str[1] = '\0';
+			RETVAL_STRING(completion_append_str);
```

Comment:
There is some inconsistency between how this is handled compared to `rl_pending_input`, where, weirdly, a string is used to set it, but we then return the character as an integer. However I think your implementation here is preferable, so let's go with it.

Reply:
If you like, this line can be changed to `RETURN_INTERNED_STR(ZSTR_CHAR(oldval))`. That would make the code both simpler and faster.